code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -55,6 +57,41 @@
 /mob/living/simple_animal/hostile/Life()
 	if(timestopped)
 		return 0 //under effects of time magick
+	if(!client || deny_client_move) //Ventcrawling stuff
+		if(entry_vent)
+			if(get_dist(src, entry_vent) <= 1)
+				if(entry_vent.network && entry_vent.network.normal_members.len)
+					var/list/vents = list()
+					for(var/obj/machinery/atmospherics/unary/vent_pump/temp_vent in entry_vent.network.normal_members)
+						if(!temp_vent.welded)
+							vents.Add(temp_vent)
+					if(!vents.len)
+						entry_vent = null
+						return
+					var/obj/machinery/atmospherics/unary/vent_pump/exit_vent = pick(vents)
+					LoseAggro()
+					spawn(rand(20,60))
+						var/travel_time = round(get_dist(loc, exit_vent.loc) / 2)
+						forceMove(exit_vent)
+						spawn(travel_time)
+
+							if(!exit_vent)
+								forceMove(entry_vent)
+								entry_vent = null
+								return
+
+							if(prob(50))
+								src.visible_message("<span class='notice'>You hear something squeezing through the ventilation ducts.</span>",2)
+							sleep(travel_time)
+
+							if(!exit_vent)
+								forceMove(entry_vent)
+								entry_vent = null
+								return
+							forceMove(exit_vent.loc)
+							entry_vent = null
+				else
+					entry_vent = null
```

Comment:
once you do as I asked and make entry_vent merely a proc variable, you don't need ANY of these where it gets set to null since the var will stop existing

Reply:
All nulls gone